chore: update instructions file (#21)
stitchng/adonis-extensions: instructions.md

const globalMiddleware = [
  'Adonis/Middleware/Session',
  ...
  'Adonis/Middleware/UpdateViewData',

>   'Adonis/Middleware/StreamResponseBody', // optional

Author: ```suggestion
```

    | JSONP Callback
    |--------------------------------------------------------------------------
    |
    | Default jsonp callback to be used when callback query string is missing
    | in request url.
    |
    */
    jsonpCallback: 'callback',

    /*
    |--------------------------------------------------------------------------
    | Etag
    |--------------------------------------------------------------------------
    |
    | Set etag on all HTTP response. In order to disable for selected routes,
    | you can call the `response.send` with an options object as follows.
    |
    | response.send('Hello', { ignoreEtag: true })
    |
    */
    etag: false

    /*
    |--------------------------------------------------------------------------
    | Http Compression
    |--------------------------------------------------------------------------
    |
    | Set compression on all HTTP responses and package them into NodeJS Streams.
    |
    | response.transform('ascii').sendToStream(Date.now())
    |
    */
    compression: {
      enabled: false,
      algo: Env.get('APP_HTTP_COMPRESSION_ALGO', 'gzip')
    }
  },
  ...
}
```

> 

Author: ```suggestion
## Using NodeJS Streams (HTTP Streaming / Compression)

>You have to include the `stream` named middleware (and optionally set AdonisJS middleware arguments: `stream: chunked, multipart`) to utilise **NodeJS Streams** on the route you wish to enable HTTP streaming and/or HTTP compression. The `chunkedResponse` config in _config/extensions.js_ is used only for app-level configurations while the middleware argument is used for route-level configurations.

```js

const Route = use('Route')

Route.get('/timestamp', function ({ response }) {
  response.sendToStream({ time: Date.now() })
}).middleware(['auth', 'stream:chunked']).as('app.timestamping')
```

    |
    | Set etag on all HTTP response. In order to disable for selected routes,
    | you can call the `response.send` with an options object as follows.
    |
    | response.send('Hello', { ignoreEtag: true })
    |
    */
    etag: false

    /*
    |--------------------------------------------------------------------------
    | Http Compression
    |--------------------------------------------------------------------------
    |
    | Set compression on all HTTP responses and package them into NodeJS Streams.
    |
    | response.transform('ascii').sendToStream(Date.now())
    |
    */
    compression: {
      enabled: false,
      algo: Env.get('APP_HTTP_COMPRESSION_ALGO', 'gzip')
    }
  },
  ...
}
```
## Using NodeJS Streams (HTTP Streaming / Compression)

>You have to include the `stream` named middleware (and optionally set AdonisJS middleware arguments: `stream: chunked, multipart`) to utilise **NodeJS Streams** on the route you wish to enable HTTP streaming and/or HTTP compression. The `chunkedResponse` config in _config/extensions.js_ is used only for app-level configurations while the middleware argument is used for route-level configurations.

```js

> 
> const Route = use('Route')

Author: ```suggestion

/** @type {typeof import('@adonisjs/framework/src/Route/Manager')} */
const Route = use('Route')
```

Route.get('/timestamp', function ({ response }) {
  response.sendToStream({ time: Date.now() })

> }).middleware(['auth', 'stream:chunked']).as('app.timestamping')

Author: ```suggestion
}).middleware(['auth', 'stream:chunked']).as('app.timestamping')

```

const globalMiddleware = [
  'Adonis/Middleware/Session',
  ...

>   'Adonis/Middleware/UpdateViewData',

Reviewer: ```suggestion
  'Adonis/Middleware/UpdateViewData'
```

]
```

## Config

> The configuration is saved inside `config/extension.js` file. Tweak it accordingly.

Reviewer: ```suggestion
The configuration for this package is saved inside `config/extension.js` file. Tweak it according to your preference. Also, you will need to update the `config/app.js` file to include a section for HTTP compression configuration. This section should go under the `http` sub key and be title `compression` as depicted below:
```

    | sudomains.
    | cheatsheet.adonisjs.com      - offset - 2
    | virk.cheatsheet.adonisjs.com - offset - 3
    |
    */
    subdomainOffset: 2,

    /*
    |--------------------------------------------------------------------------
    | JSONP Callback
    |--------------------------------------------------------------------------
    |
    | Default jsonp callback to be used when callback query string is missing
    | in request url.
    |
    */
    jsonpCallback: 'callback',

    /*
    |--------------------------------------------------------------------------
    | Etag
    |--------------------------------------------------------------------------
    |
    | Set etag on all HTTP response. In order to disable for selected routes,
    | you can call the `response.send` with an options object as follows.
    |
    | response.send('Hello', { ignoreEtag: true })
    |
    */
    etag: false

    /*
    |--------------------------------------------------------------------------
    | Http Compression
    |--------------------------------------------------------------------------
    |
    | Set compression on all HTTP responses and package them into NodeJS Streams.
    |

>     | response.transform('ascii').sendToStream(Date.now())

Reviewer: ```suggestion
    | response.transform('ascii', { chunked: false, multipart: false }).sendToStream(Date.now())
```

    |
    */
    compression: {
      enabled: false,
      algo: Env.get('APP_HTTP_COMPRESSION_ALGO', 'gzip')
    }
  },
  ...
}
```
## Using NodeJS Streams (HTTP Streaming / Compression)

>You have to include the `stream` named middleware (and optionally set AdonisJS middleware arguments: `stream: chunked, multipart`) to utilise **NodeJS Streams** on the route you wish to enable HTTP streaming and/or HTTP compression. The `chunkedResponse` config in _config/extensions.js_ is used only for app-level configurations while the middleware argument is used for route-level configurations.

```js

/** @type {typeof import('@adonisjs/framework/src/Route/Manager')} */
const Route = use('Route')

Route.get('/timestamp', function ({ response }) {

>   response.sendToStream({ time: Date.now() })
> }).middleware(['auth', 'stream:chunked']).as('app.timestamping')

Reviewer: ```suggestion
  response.sendToStream({ time: Date.now() }, 'application/json; charset=utf-8')
}).middleware(['auth', 'stream: chunked, mutipart']).as('app.timestamping')

Route.post('/random', function ({ response }) {
  response.sendToStream(Math.random() * 12345)
}).middleware('auth', 'stream: chunked']).as('app.random')
```

    |
    | Default jsonp callback to be used when callback query string is missing
    | in request url.
    |
    */
    jsonpCallback: 'callback',

    /*
    |--------------------------------------------------------------------------
    | Etag
    |--------------------------------------------------------------------------
    |
    | Set etag on all HTTP response. In order to disable for selected routes,
    | you can call the `response.send` with an options object as follows.
    |
    | response.send('Hello', { ignoreEtag: true })
    |
    */
    etag: false

    /*
    |--------------------------------------------------------------------------
    | Http Compression
    |--------------------------------------------------------------------------
    |
    | Set compression on all HTTP responses and package them into NodeJS Streams.
    |
    | response.transform('ascii', { chunked: false, multipart: false }).sendToStream(Date.now())
    |
    */
    compression: {
      enabled: false,
      algo: Env.get('APP_HTTP_COMPRESSION_ALGO', 'gzip')
    }
  },
  ...
}
```
## Using NodeJS Streams (HTTP Streaming / Compression)

> 

Reviewer: ```suggestion

As it concerns **AdonisJS** 4, [NodeJS Streams](https://nodesource.com/blog/understanding-streams-in-nodejs/) are only used when files (usually on disk) are to be sent to the (HTTP) client (e.g. web, mobile) via either the `response.download()` or `response.attachment()` methods (on the AdonisJS response objects).  This has made it such that dynamically-created data (e.g. from a database as JSON) that is large enough can't utilize [NodeJS Streams](https://nodesource.com/blog/understanding-streams-in-nodejs/). Now, with **adonisjs-extensions**, you can wield the power of [NodeJS Streams](https://nodesource.com/blog/understanding-streams-in-nodejs/) while sending dynamically-created data to the (HTTP) client without any hindrance.

Below is how to get started using NodeJS streams:

```

>You have to include the `stream` named middleware (and optionally set AdonisJS middleware arguments: `stream: chunked, multipart`) to utilise **NodeJS Streams** on the route you wish to enable HTTP streaming and/or HTTP compression. The `chunkedResponse` config in _config/extensions.js_ is used only for app-level configurations while the middleware argument is used for route-level configurations.

```js

/** @type {typeof import('@adonisjs/framework/src/Route/Manager')} */
const Route = use('Route')

Route.get('/timestamp', function ({ response }) {
  response.sendToStream({ time: Date.now() }, 'application/json; charset=utf-8')
}).middleware(['auth', 'stream: chunked, mutipart']).as('app.timestamping')

Route.post('/random', function ({ response }) {
  response.sendToStream(Math.random() * 12345)

> }).middleware('auth', 'stream: chunked']).as('app.random')

Reviewer: ```suggestion
}).middleware(['stream: chunked']).as('app.randoming')
```
